owiddata/generate-owiddata-stats.py

```diff
@@ -0,0 +1,77 @@
+import argparse
+import datetime
+import json
+from datetime import date
+import os
+import pandas as pd
+
+def convert_date(git_date):
+    '''Reformat git commit style datetimes (ISO 8601) to Month DD, YYYY.
+    Throws a ValueError if git_date cannot be parsed as an ISO 8601 datetime.
+    '''
+    # 'Z' indicates Coordinated Universal Time (UTC)
+    # Replace with '+00:00', which is the UTC representation recognized
+    # by the parser
+    # https://en.wikipedia.org/wiki/ISO_8601#Coordinated_Universal_Time_(UTC)
+    git_date = git_date.replace('Z', '+00:00')
+
+    # Remove the leading zero of the day
+    # Assumes the year will not begin with 0
+    return datetime.datetime.fromisoformat(git_date).strftime('%B %d, %Y').replace(' 0', ' ')
+
+def main(args):
+    # Import data from github.com/owid/covid-19-data
+    owid_stats = dict()
+
+    if('OWID_COMMIT_SHA' in os.environ):
+        owid_stats['owid_commit_sha'] = os.environ['OWID_COMMIT_SHA']
+    if('OWID_COMMIT_DATE' in os.environ):
+        owid_stats['owid_commit_date'] = os.environ['OWID_COMMIT_DATE']
+        owid_stats['owid_commit_date_pretty'] = convert_date(os.environ['OWID_COMMIT_DATE'])
+
+    # Retrieve data from OWID
+    locations_url = 'https://raw.githubusercontent.com/owid/covid-19-data/master/public/data/vaccinations/locations.csv'
```

Comment:
For `locations_url`, `numbers_url`, and `manufacturer_url` I think we should download from a specific commit instead of `master`.

I had been downloading these in the wrapper shell script, but this is slightly better to do it in the Python script if we can lock the versioning.

Reply:
That sounds great to me! They seem to update once per day, so it seems like we would definitely only need to rerun once per day. I'm not sure how to do this, though. I assume there's some way to ask for the most recent commit?